pkg/prometheus/promcfg.go

```diff
@@ -4092,3 +4044,24 @@ func getLowerByteSize(v *monitoringv1.ByteSize, cpf *monitoringv1.CommonPromethe
 func isByteSizeEmpty(v *monitoringv1.ByteSize) bool {
 	return v == nil || *v == ""
 }
+
+func (cg *ConfigGenerator) addFiltersToYaml(cfg yaml.MapSlice, filters []monitoringv1alpha1.Filter) yaml.MapSlice {
+	if len(filters) == 0 {
+		return cfg
+	}
+
+	filtersYamlMap := []yaml.MapSlice{}
+	for _, filter := range filters {
```

Comment:
to have a generated config which is 100% deterministic and stable, I'd sort the filters slice by name and then iterate over the sorted names.

Reply:
I thought of this too. Should we sort all the fields for all the generated SDs as well?